modular_skyrat/modules/modular_items/lewd_items/code/lewd_clothing/shibari_worn.dm

```diff
@@ -0,0 +1,330 @@
+
+	greyscale_config = /datum/greyscale_config/shibari_clothes/fullbody
+	greyscale_config_worn = /datum/greyscale_config/shibari_worn/fullbody
+	greyscale_config_worn_digi = /datum/greyscale_config/shibari_worn_digi/fullbody
+	greyscale_config_worn_taur_snake = /datum/greyscale_config/shibari_worn_taur_snake/fullbody
+	greyscale_config_worn_taur_paw = /datum/greyscale_config/shibari_worn_taur_paw/fullbody
+	greyscale_config_worn_taur_hoof = /datum/greyscale_config/shibari_worn_taur_hoof/fullbody
+	greyscale_colors = "#bd8fcf#bd8fcf"
+
+//processing stuff
+/obj/item/clothing/under/shibari/full/process(delta_time)
+	. = ..()
+	var/mob/living/carbon/human/hooman = loc
+	if(tightness == SHIBARI_TIGHTNESS_LOW && hooman.pleasure< 20)
+		hooman.adjustPleasure(0.6 * delta_time)
+	if(tightness == SHIBARI_TIGHTNESS_MED && hooman.pleasure < 60)
+		hooman.adjustPleasure(0.6 * delta_time)
+	if(tightness == SHIBARI_TIGHTNESS_HIGH)
+		hooman.adjustPleasure(0.6 * delta_time)
+		if(hooman.pain < 40)
+			hooman.adjustPain(0.6 * delta_time)
+		if(prob(10))
+			hooman.adjustOxyLoss(5)
+
+/obj/item/clothing/gloves/shibari_hands
+	name = "shibari arms bondage"
+	desc = "Bondage ropes that cover arms."
+	icon_state = "shibari_arms"
+	icon = 'modular_skyrat/modules/modular_items/lewd_items/icons/obj/lewd_clothing/lewd_gloves.dmi'
+	worn_icon = 'modular_skyrat/modules/modular_items/lewd_items/icons/mob/lewd_clothing/lewd_gloves.dmi'
+	body_parts_covered = NONE
+	strip_delay = 100
+	item_flags = DROPDEL
+
+	greyscale_config = /datum/greyscale_config/shibari_clothes/hands
+	greyscale_config_worn = /datum/greyscale_config/shibari_worn/hands
+	greyscale_colors = "#bd8fcf"
+
+	///should this clothing item use the emissive system
+	var/glow = FALSE
+
+/obj/item/clothing/gloves/shibari_hands/update_overlays()
+	. = ..()
+	if(glow)
+		. += emissive_appearance(icon, icon_state, alpha = 100)
+
+/obj/item/clothing/gloves/shibari_hands/worn_overlays(mutable_appearance/standing, isinhands, icon_file)
+	. = ..()
+	if(glow)
+		. += emissive_appearance(standing.icon, standing.icon_state, alpha = 100)
+
+/obj/item/clothing/gloves/shibari_hands/Destroy()
+	var/mob/living/carbon/human/hooman = loc
+	if(HAS_TRAIT(hooman, TRAIT_ROPEBUNNY))
+		hooman.remove_status_effect(/datum/status_effect/ropebunny)
+	for(var/obj/item in contents)
+		item.forceMove(get_turf(src))
+	. = ..()
+
+/obj/item/clothing/gloves/shibari_hands/equipped(mob/user, slot)
+	. = ..()
+	RegisterSignal(src, COMSIG_ATOM_ATTACK_HAND, .proc/handle_take_off, user)
+
+
+/obj/item/clothing/gloves/shibari_hands/proc/handle_take_off(datum/source, mob/user)
+	SIGNAL_HANDLER
+	INVOKE_ASYNC(src, .proc/handle_take_off_async, user)
+	return COMPONENT_CANCEL_ATTACK_CHAIN
+
+/obj/item/clothing/gloves/shibari_hands/proc/handle_take_off_async(mob/user)
+	if(iscarbon(user))
+		var/mob/living/carbon/human/hooman = user
+		if(do_after(hooman, HAS_TRAIT(hooman, TRAIT_RIGGER) ? 2 SECONDS : 10 SECONDS, target = src))
+			qdel(src)
```

Comment:
This is the third time I've seen this code. Please fix this copypaste.

Reply:
different subtypes, no way to do it without making an element, which is unnecessary when this is literally the only case that uses it

Reply:
This level of copypaste is really, really bad. There's definitely a way to clean it up.

Reply:
three times isn’t bad, blame someone else for hands being a different subtype